src/offer/service.js

```diff
@@ -0,0 +1,96 @@
+const axios = require("axios");
+const fs = require("fs");
+const path = require("path");
+
+class OfferService {
+  constructor(apiUrl) {
+    this.offerUrl = `${apiUrl}/offering/offer/`;
+    this.offersFilePath = path.resolve("./offer-bundle.example.json");
+    this.signatureService = require("../helpers/signer.service").init(
+      process.env.KEY
+    );
+  }
+
+  async createOffer() {
+    try {
+      const offerDataset = JSON.parse(
+        fs.readFileSync(this.offersFilePath, { encoding: "utf-8" })
+      );
+      const response = await axios.post(this.offerUrl, offerDataset["create"], {
+        headers: {
+          "x-publisher-token": process.env.PUBLISHER_TOKEN,
+          signature: this.signatureService.createSignature(
+            offerDataset["create"]
+          ),
+        },
+      });
+      const responseBody = response.data;
+      await this.updateOfferId();
+
+      return responseBody;
+    } catch (error) {
+      if (error.response) {
+        return error.response.data;
+      } else {
+        console.error(error);
+        return { error: "Internal Server Error" };
+      }
+    }
+  }
+
+  async updateOffer() {
+    try {
+      const updateOfferDataset = JSON.parse(
+        fs.readFileSync(this.offersFilePath, { encoding: "utf-8" })
+      )["update"];
+      const offerId = updateOfferDataset["publisherOfferId"];
+      const modifiedOfferDataset = this.removeFields(updateOfferDataset, [
+        "publisherOfferId",
+        "createdBy",
+        "intervals",
+      ]);
+      const response = await axios.put(
+        this.offerUrl + offerId,
+        modifiedOfferDataset,
```

Comment:
looks complicated to understand what the object should look like, maybe we should put plain objects instead